src/app/pages/swap/components/swap-details/swap-details.tsx

```diff
@@ -1,14 +1,87 @@
+import { HStack, styled } from 'leather-styles/jsx';
+
+import { createMoney } from '@shared/models/money.model';
+import { isDefined, isUndefined } from '@shared/utils';
+
+import { formatMoneyPadded } from '@app/common/money/format-money';
+import { microStxToStx } from '@app/common/money/unit-conversion';
+import { getEstimatedConfirmationTime } from '@app/common/transactions/stacks/transaction.utils';
+import { ChevronUpIcon } from '@app/components/icons/chevron-up-icon';
+import { SwapSubmissionData, useSwapContext } from '@app/pages/swap/swap.context';
+import { useStacksBlockTime } from '@app/query/stacks/info/info.hooks';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+
 import { SwapDetailLayout } from './swap-detail.layout';
 import { SwapDetailsLayout } from './swap-details.layout';
 
-// TODO: Replace with live data
+function RouteNames(props: { swapSubmissionData: SwapSubmissionData }) {
+  return props.swapSubmissionData.router.map((route, i) => {
+    const insertIcon = isDefined(props.swapSubmissionData.router[i + 1]);
+    return (
+      <>
+        <styled.span>{route.name}</styled.span>
+        {insertIcon && <ChevronUpIcon transform="rotate(90)" />}
+      </>
+    );
+  });
+}
+
 export function SwapDetails() {
+  const { swapSubmissionData } = useSwapContext();
+  const { isTestnet } = useCurrentNetworkState();
+  const { data: blockTime } = useStacksBlockTime();
+
+  if (
+    isUndefined(swapSubmissionData) ||
+    isUndefined(swapSubmissionData.swapAssetFrom) ||
+    isUndefined(swapSubmissionData.swapAssetTo)
+  )
+    return null;
+
   return (
     <SwapDetailsLayout>
-      <SwapDetailLayout title="Placeholder" tooltipLabel="Tooltip info" value="0" />
-      <SwapDetailLayout title="Placeholder" value="0" />
-      <SwapDetailLayout title="Placeholder" value="0" />
-      <SwapDetailLayout title="Placeholder" value="0" />
+      <SwapDetailLayout title="Protocol" value={swapSubmissionData.protocol} />
+      <SwapDetailLayout
+        title="Route"
+        value={
+          <HStack gap="space.01">
+            <RouteNames swapSubmissionData={swapSubmissionData} />
+          </HStack>
+        }
+      />
+      <SwapDetailLayout
+        title="Minimum received"
+        value={formatMoneyPadded(
+          createMoney(
+            Number(swapSubmissionData.swapAmountTo) * (1 - swapSubmissionData.slippage),
```

Comment:
Can we perform all mathematical operations outside JSX? It's important logic that shouldn't get lost among markup.

Also, we should **always** use BigNumber when doing maths on monetary data.